ocaml/xcp-rrdd/bin/rrdd/xcp_rrdd.ml

```diff
@@ -347,6 +347,39 @@ let dss_loadavg () =
     )
   ]
 
+let dss_hostload xc domains =
+  let physinfo = Xenctrl.physinfo xc in
+  let pcpus = physinfo.Xenctrl.nr_cpus in
+  let rec sum acc n f =
+    match n with n when n >= 0 -> sum (acc + f n) (n - 1) f | _ -> acc
+  in
+  let load =
+    List.fold_left
+      (fun acc (dom, _, domid) ->
+        sum 0 dom.Xenctrl.max_vcpu_id (fun id ->
+            let vcpuinfo = Xenctrl.domain_get_vcpuinfo xc domid id in
+            if vcpuinfo.Xenctrl.online && not vcpuinfo.Xenctrl.blocked then
+              1
+            else
+              0
```

Comment:
A minor observation: adding with 0 seems not necessary, could just return the `acc` here.

Reply:
Confirmed with Ming, won't change.